src/agent/src/device.rs

```diff
@@ -717,20 +758,48 @@ fn split_vfio_pci_option(opt: &str) -> Option<(&str, &str)> {
     }
 }
 
+// VFIO device could be PCI or AP, which requires different handling. The first option is checked
+// for the prefix, `pci:` or `ap:` (both within one device are not possible).
+#[instrument]
```

Comment:
Demultiplexing here seems silly to me.  Why not just create new device type strings for AP devices (and rename the existing PCI ones if we want)?